packages/react/popper/src/Popper.tsx

```diff
@@ -48,15 +53,11 @@ const PopperAnchor = React.forwardRef((props, forwardedRef) => {
   const { virtualRef, children, ...anchorProps } = props;
   const context = usePopperContext(ANCHOR_NAME);
   const ref = React.useRef<React.ElementRef<typeof Primitive>>(null);
-  const composedRefs = useComposedRefs(
-    forwardedRef,
-    ref,
-    context.anchorRef as React.MutableRefObject<React.ElementRef<typeof Primitive>>
-  );
+  const composedRefs = useComposedRefs(forwardedRef, ref, (node) => context.onAnchorChange(node));
 
   React.useEffect(() => {
     if (virtualRef?.current) {
-      context.anchorRef.current = virtualRef.current;
+      context.onAnchorChange(virtualRef.current);
```

Comment:
Am I reading this correctly here that if you passed a `virtualRef`, `onAnchorChange` will get  called twice potentially, first when the node mounts, set to the node, and then overriden in the effect here?

edit: actually after expanding the code I realised that the composedRef wouldn't be called if a virtualRef is passed obviously. It's a bit annoying how the code is split here. Not sure if we can improve this.

Reply:
Oh wait, now that we're using nodes, I might be able to improve this again by doing:

```jsx
const ref = React.useRef();
const composedRefs = (forwardedRef, ref);

React.useEffect(() => {
	context.onAnchorChange(virtualRef ? virtualRef.current : ref.current);
});
```

Let me check...

Reply:
Yep works... it broke the reordered parts stories before but all seems to be good now we're using nodes 👍